connections/http.go

```diff
@@ -44,13 +45,13 @@ func HTTPGetWithClient[T protoreflect.ProtoMessage](url string, client *http.Cli
 	return nil
 }
 
-func HTTPPostWithClient[T protoreflect.ProtoMessage](url string, client *http.Client, body interface{}, val T, authHeader string) error {
+func HTTPPostWithClient[T protoreflect.ProtoMessage](ctx context.Context, url string, client *http.Client, body interface{}, val T, authHeader string) error {
 	b, err := json.Marshal(body)
 	if err != nil {
 		return err
 	}
 
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(b))
+	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(b))
```

Comment:
How does the context get sent with the request?

Reply:
And we are checking the timeout of the context for each request on the API side?

Reply:
it's purely a client side timeout. So the SDK is the one that can set a timeout for the server to respond properly in, and if it doesn't then the SDK will return an error